README.md

```diff
@@ -261,82 +271,56 @@ Renders the widget to the DOM, and passes control back to your app through succe
+- `success` *(optional)* - Function that is called when the user has completed an authentication flow. If an [OpenID Connect redirect flow](#openid-connect) is used, this function can be omitted. See below for more details.
 - `error` *(optional)* - Function that is called when the widget has been initialized with invalid config options, or has entered a state it cannot recover from. If omitted, a default function is used to output errors to the console.
 
+#### success
+
+Function that handles non-error events, including the completion of a successful authentication flow by a user. Also handles other flows such as reset password and self-registration flows.
+
+_OIDC vs Non-OIDC Configurations:_
+
+New integrations are recommended to use OIDC as it uses a more lightweight REST-based protocol and has more widespread usage. Less common is the need for non-OIDC integrations in self-hosted applications, however one example of this usage is when using the Okta-hosted sign-in widget to redirect a user to the Okta Dashboard after authentication.
+
+**`res` parameter:**
+
+This `success` function will be called with an object, that can have various properties, depending on how the widget is configured:
+
 ```javascript
-signIn.renderEl(
-  // Assumes there is an empty element on the page with an id of 'osw-container'
-  {el: '#osw-container'},
-
-  function success(res) {
-    // The properties in the response object depend on two factors:
-    // 1. The type of authentication flow that has just completed, determined by res.status
-    // 2. What type of token the widget is returning
-
-    // The user has started the password recovery flow, and is on the confirmation
-    // screen letting them know that an email is on the way.
-    if (res.status === 'FORGOT_PASSWORD_EMAIL_SENT') {
-      // Any followup action you want to take
-      return;
-    }
+success({
+  status: String,
+  username: Optional<String>,
+  activationToken: Optional<Object>,
+  tokens: Optional<Object>,
+  type: Optional<String>,
+  user: Optional<Object>,
+  stepUp: Optional<Function>,
+  session: Optional<Object>,
+  next: Optional<Function>,
+})
+```
 
-    // The user has started the unlock account flow, and is on the confirmation
-    // screen letting them know that an email is on the way.
-    if (res.status === 'UNLOCK_ACCOUNT_EMAIL_SENT') {
-      // Any followup action you want to take
-      return;
-    }
+- `status` *(string)* - Always present. One of: `FORGOT_PASSWORD_EMAIL_SENT`, `UNLOCK_ACCOUNT_EMAIL_SENT`, `ACTIVATION_EMAIL_SENT`, `REGISTRATION_COMPLETE`, or `SUCCESS`
 
-    // The user has successfully completed the authentication flow
-    if (res.status === 'SUCCESS') {
+- `username` *(optional\<string\>)* - Only present when `status` is one of `FORGOT_PASSWORD_EMAIL_SENT`, `UNLOCK_ACCOUNT_EMAIL_SENT`, `ACTIVATION_EMAIL_SENT`, or `REGISTRATION_COMPLETE`.
```

Comment:
a bit inconsistency:
`optional\<string\>` vs `Optional\<String\>`

No preference but just make it consistent.

Reply:
Ah yeah, I'll go with lowercase ones since the previous docs have it that way.